pkg/cli/token_test.go

```diff
@@ -15,190 +15,176 @@
+			token, err := jwt.ParseString(tokenStr,
+				jwt.WithVerify(false),
+				jwt.WithValidate(false),
+				jvspb.WithTypedJustifications())
+			if err != nil {
+				t.Fatal(err)
+			}
 
-			err := runTokenCmd(cmd, nil)
-			if diff := testutil.DiffErrString(err, tc.wantErr); diff != "" {
-				t.Errorf("unexpected err: %s", diff)
+			// Validate standard claims.
+			if got, want := token.Audience(), []string{justification.DefaultAudience}; !reflect.DeepEqual(got, want) {
+				t.Errorf("aud: expected %q to be %q", got, want)
+			}
+			if got := token.Expiration(); !got.After(now) {
+				t.Errorf("exp: expected %q to be after %q (%q)", got, now, got.Sub(now))
+			}
+			if got := token.IssuedAt(); got.IsZero() {
+				t.Errorf("iat: expected %q to be", got)
+			}
+			if got, want := token.Issuer(), "jvsctl"; got != want {
+				t.Errorf("iss: expected %q to be %q", got, want)
+			}
+			if got, want := token.Subject(), "jvsctl"; got != want {
+				t.Errorf("sub: expected %q to be %q", got, want)
 			}
 
-			if gotToken := buf.String(); gotToken != tc.wantToken {
-				t.Errorf("justification token got=%q, want=%q", gotToken, tc.wantToken)
+			// Validate custom claims.
+			justifications, err := jvspb.GetJustifications(token)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if diff := cmp.Diff(tc.expJustifications, justifications, cmpopts.IgnoreUnexported(jvspb.Justification{})); diff != "" {
+				t.Errorf("justs: diff (-want, +got):\n%s", diff)
 			}
 		})
 	}
 }
 
-func TestRunTokenCmd_Breakglass(t *testing.T) {
-	// Cannot parallel because the global CLI config.
-	// These are global flags.
-	cfg = &config.CLIConfig{
-		Server: "example.com",
-		Authentication: &config.CLIAuthentication{
-			Insecure: true,
-		},
-	}
-	flagBreakglass = true
-	flagTokenExplanation = "i-have-reason"
-	flagTTL = time.Minute
-
-	// Override timeFunc to have fixed time for test.
-	now := time.Now().UTC()
-	t.Cleanup(testRunTokenCmdCleanup)
-
-	var buf bytes.Buffer
-	cmd := &cobra.Command{}
-	cmd.SetArgs([]string{"--iat", strconv.FormatInt(now.Unix(), 10)})
-	cmd.SetOut(&buf)
-
-	if err := runTokenCmd(cmd, nil); err != nil {
-		t.Fatal(err)
-	}
+type fakeJVS struct {
+	jvspb.UnimplementedJVSServiceServer
+	returnErr error
+}
 
-	// Validate message headers - we have to parse the full envelope for this.
-	message, err := jws.Parse(buf.Bytes())
-	if err != nil {
-		t.Fatal(err)
-	}
-	if sigs, want := message.Signatures(), 1; len(sigs) != want {
-		t.Errorf("expected length %d to be %d: %#v", len(sigs), want, sigs)
-	} else {
-		headers := sigs[0].ProtectedHeaders()
-		if got, want := headers.Type(), "JWT"; got != want {
-			t.Errorf("typ: expected %q to be %q", got, want)
-		}
-		if got, want := string(headers.Algorithm()), "HS256"; got != want {
-			t.Errorf("alg: expected %q to be %q", got, want)
-		}
+func (j *fakeJVS) CreateJustification(_ context.Context, req *jvspb.CreateJustificationRequest) (*jvspb.CreateJustificationResponse, error) {
```

Comment:
This is beyond my bar :) Since we're returning whatever token the server returns as a string, I would not really generate a real JTW.

Reply:
The reason I did this is so we can use the same table-driven tests for testing breakglass tokens and testing tokens. Otherwise the testing logic becomes branching and that's less than ideal.